beamer/agent/events.py

```diff
@@ -67,9 +67,11 @@ class TxEvent(Event):
 
 
 @dataclass(frozen=True)
-class FinalityPeriodUpdated(TxEvent, SourceChainEvent):
-    target_chain_id: ChainId
+class ChainUpdated(TxEvent, SourceChainEvent):
+    chain_id: ChainId
     finality_period: int
+    target_weight_p_p_m: int
```

Comment:
Nitpick: I think this would be more readable as `target_weight_ppm`

Reply:
We can have this field named `target_weight_ppm` more nicely by improving the `_camel_to_snake` function in the same module:

```python
def _camel_to_snake(s: str) -> str:                                                 
    snake = "".join(                                                                
        "%c_" % current if current.islower() and next.isupper() else current.lower()
        for current, next in pairwise(s)                                            
    )                                                                               
    # The last character isn't included in snake due to how pairwise() works,       
    # so add it manually.                                                           
    return snake.lstrip("_") + s[-1].lower()                                        
```

(Needs `from itertools import pairwise` at the top.)

Reply:
can you please create its own PR for it once we have it merged?

Reply:
> can you please create its own PR for it once we have it merged?

No problem, man. :+1: